### PR TITLE
Add verify-goimport to verify all import items are well sorted

### PR DESCRIPTION
**Pull request by HanningLin**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

PR fixing this issue https://github.com/etcd-io/etcd/issues/15469

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -632,6 +632,19 @@ function genproto_pass {
   "${ETCD_ROOT_DIR}/scripts/verify_genproto.sh"
 }
 
+function goimport_for_module {
+  GOFILES=$(run go list  --f "{{with \$d:=.}}{{range .GoFiles}}{{\$d.Dir}}/{{.}}{{\"\n\"}}{{end}}{{end}}" ./...)
+  TESTGOFILES=$(run go list  --f "{{with \$d:=.}}{{range .TestGoFiles}}{{\$d.Dir}}/{{.}}{{\"\n\"}}{{end}}{{end}}" ./...)
+  cd "${ETCD_ROOT_DIR}/tools/mod"
+  FILESNEEDSFIX=$(echo "${GOFILES}" "${TESTGOFILES}" | grep -v '.gw.go' | grep -v '.pb.go' | xargs -n 100 go run golang.org/x/tools/cmd/goimports -l -local go.etcd.io)
```
Comment (jmhbnz) [inline scripts/test.sh:639]: Thanks for raising this @HanningLin!

I'm curious about the `-n 100` for xargs, wouldn't that mean the the full output of files needing to be fixed wouldn't be printed if it was more than 100 files?  If so can we just leave that flag off?

Comment (ahrtr) [reply scripts/test.sh:639]: It should be fine. It means at most 100 files per command line.


```
       -n max-args, --max-args=max-args
              Use at most max-args arguments per command line.  Fewer
              than max-args arguments will be used if the size (see the
              -s option) is exceeded, unless the -x option is given, in
              which case xargs will exit.
```

The script is copied from [fix.sh#L34-L39](https://github.com/etcd-io/etcd/blob/c6d8b65ea7a77556f44a826eecf08c83cc966bca/scripts/fix.sh#L34-L39). The only minor comment is duplication.

File: scripts/test.sh [hunk 2]
```diff
@@ -632,6 +632,19 @@ function genproto_pass {
...
+  if [ -n "$FILESNEEDSFIX" ]; then
+    log_error -e "the following files are not sync with 'goimports'. run 'make fix'\\n$FILESNEEDSFIX"
+    return 255
+  fi
+}
+function goimport_pass {
```
Comment (ahrtr) [inline scripts/test.sh:645]: nit: please add a blank line before and after this function.

File: scripts/test.sh (hunk 1 above)
Comment (jmhbnz) [reply scripts/test.sh:639]: Ahh right, xargs will split and sequentially send to  `go run` ok in that case this all looks good to me.

File: scripts/test.sh (hunk 2 above)
Comment (HanningLin) [reply scripts/test.sh:646]: a new line was added

File: scripts/test.sh (hunk 1 above)
Comment (HanningLin) [reply scripts/test.sh:639]: Yes, this is mostly copied from fix.sh but with a minor diff using `-l` to list the files

File: scripts/test.sh [hunk 3]
```diff
@@ -632,6 +632,20 @@ function genproto_pass {
...
+  FILESNEEDSFIX=$(echo "${GOFILES}" "${TESTGOFILES}" | grep -v '.gw.go' | grep -v '.pb.go' | xargs -n 100 go run golang.org/x/tools/cmd/goimports -l -local go.etcd.io)
+  if [ -n "$FILESNEEDSFIX" ]; then
+    log_error -e "the following files are not sync with 'goimports'. run 'make fix'\\n$FILESNEEDSFIX"
+    return 255
+  fi
+}
+
+function goimport_pass {
+  run_for_modules goimport_for_module
+}
```
Comment (ahrtr) [inline scripts/test.sh:648]: nit: please add a new line after the function as well. Please also squash the commits into just one commit.

Comment (HanningLin) [reply scripts/test.sh:648]: added and squashed

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @HanningLin

**Comment by HanningLin** [comment]
> LGTM
> 
> Thank you @HanningLin

Thanks for the review!

**Review by jmhbnz — APPROVED** [review]
/lgtm

**Review by serathius — APPROVED** [review]

**Pull request merged**